python/cudf/cudf/testing/_utils.py

```diff
@@ -40,6 +40,9 @@
 ALL_TYPES = sorted(list(dtypeutils.ALL_TYPES))
 
 
+# TODO: This function should be removed. Anywhere that it is being used should
+# instead be generating a random boolean array (bytemask) and use the public
+# APIs to set those elements to None.
```

Comment:
I have done some related (unfinished) work in this PR. I need to do some thinking on how to best refactor that draft for performance and readability. https://github.com/rapidsai/cudf/pull/10034

Reply:
Ah my apologies, I did not see that PR (perhaps since it's a draft?). I didn't mean to step on your toes there. IMO tests shouldn't rely on internal APIs any more than users should, and the way for a user to apply a mask is via a bytemask `series[bytemask] = None`, so I think it's fine for tests to also use that approach even if it is less efficient. I think this PR can remove `set_mask` but change the tests to use the internal column APIs (which is what I've done so far), and then you can make the subsequent changes to the tests in #10034 once we've come to a consensus (in case you disagree with my assertion above). Sound good?

Reply:
No worries! I started that draft but it was low priority. It might be helpful to have a cudf.testing function to assign a random null mask, since that pattern is repeated so many times in the tests. I agree that we should avoid internal APIs in tests to the greatest extent possible.